app/k9mail/src/main/java/com/fsck/k9/account/AccountLoader.kt

```diff
@@ -0,0 +1,58 @@
+package com.fsck.k9.account
+
+import app.k9mail.feature.account.common.AccountCommonExternalContract
+import app.k9mail.feature.account.common.domain.entity.Account
+import app.k9mail.feature.account.common.domain.entity.AccountOptions
+import com.fsck.k9.Preferences
+import com.fsck.k9.logging.Timber
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import com.fsck.k9.Account as K9Account
+
+class AccountLoader(
+    private val preferences: Preferences,
+    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : AccountCommonExternalContract.AccountLoader {
+
+    @Suppress("TooGenericExceptionCaught")
+    override suspend fun loadAccount(accountUuid: String): Account? {
+        return try {
+            withContext(coroutineDispatcher) {
+                load(accountUuid)
+            }
+        } catch (e: Exception) {
+            Timber.e(e, "Error while loading account")
+
+            null
+        }
+    }
+
+    private fun load(accountUuid: String): Account? {
+        val existingAccount = preferences.getAccount(accountUuid)
+
+        return if (existingAccount != null) {
+            mapToAccount(existingAccount)
+        } else {
+            null
+        }
+    }
+
+    private fun mapToAccount(account: K9Account): Account {
+        return Account(
+            uuid = account.uuid,
+            emailAddress = account.email,
+            incomingServerSettings = account.incomingServerSettings,
+            outgoingServerSettings = account.outgoingServerSettings,
+            authorizationState = account.oAuthState,
+            options = AccountOptions(
+                accountName = account.name ?: "Undefined",
+                displayName = account.senderName ?: "Undefined",
+                emailSignature = account.signature ?: "",
```

Comment:
For generic account loading most of this logic would need to be fixed. But for editing server settings we don't actually care about these values.

I suggested changing the interfaces used by `*ServerSettingsViewModel` so we don't have to load data that we don't need. I don't think this has to be changed right away. But at the very least we should add a comment stating that if a caller intends to use `AccountOptions` this method needs to be fixed.

To make this more obvious we could do the following:

```kotlin
            options = AccountOptions(
                accountName = "should not be used",
                displayName = "should not be used",
                emailSignature = "should not be used",
                checkFrequencyInMinutes = -1,
                showNotification = false,
                messageDisplayCount = 10,
            ),
```

Reply:
I think the mistage was to use the `Account` class that has all fields set as mandatory. As we anyway just need the `AccountState`, I'll change it to that and also remove the unesesarry mapping.